public/app/features/trails/MetricSelect/MetricSelectSceneReact.tsx

```diff
@@ -0,0 +1,328 @@
+            className={styles.displayOption}
+          >
+            <Select
+              value={metricPrefix}
+              onChange={model.onPrefixFilterChange}
+              onOpenMenu={() => model.reportPrefixFilterInteraction(true)}
+              onCloseMenu={() => model.reportPrefixFilterInteraction(false)}
+              options={[
+                {
+                  label: 'All metric names',
+                  value: METRIC_PREFIX_ALL,
+                },
+                ...Array.from(rootGroup?.groups.keys() ?? []).map((g) => ({ label: `${g}_`, value: g })),
+              ]}
+            />
+          </Field> */}
+        </div>
+        {/* {metricNamesError && (
+          <Alert title="Unable to retrieve metric names" severity="error">
+            <div>We are unable to connect to your data source. Double check your data source URL and credentials.</div>
+            <div>({metricNamesError})</div>
+          </Alert>
+        )}
+        {metricNamesWarning && !warningDismissed && (
+          <Alert
+            title="Unable to retrieve all metric names"
+            severity="warning"
+            onSubmit={dismissWarning}
+            onRemove={dismissWarning}
+          >
+            <div>{metricNamesWarning}</div>
+          </Alert>
+        )} */}
+        {metricNamesLoading && <div>Loading metric names...</div>}
+        <VizGridLayout>
+          {metricNamesFiltered?.map((metric, index) => (
+            <MetricSelectPanel metric={metric} selectScene={model} index={index} key={metric} />
+            // <div key={metric}>{metric}</div>
+          ))}
+        </VizGridLayout>
+      </div>
+    );
+  };
+}
+
+interface MetricSelectPanelProps {
+  metric: string;
+  selectScene: MetricSelectSceneReact;
+  index: number;
+}
+
+function MetricSelectPanel({ metric, selectScene, index }: MetricSelectPanelProps) {
+  const autoQuery = getAutoQueriesForMetric(metric);
+  const filters = getFilters(selectScene);
+  const currentFilterCount = filters?.length || 0;
+
+  const queries = autoQuery.preview.queries.map((query) =>
+    convertPreviewQueriesToIgnoreUsage(query, currentFilterCount)
+  );
+
+  // Instead of having to wrap each viz in a nested variable context I think this is simpler and faster
+  // This is a bit hacky and should be replaced a bit more robust implementation
+  queries.forEach((query) => {
+    query.expr = query.expr?.replace('${metric}', metric);
+    query.legendFormat = query.legendFormat?.replace('${metric}', metric);
+  });
+
+  const dataProvider = useQueryRunner({
+    datasource: trailDS,
+    maxDataPoints: MDP_METRIC_PREVIEW,
+    queries,
+    cacheKey: [metric],
+  });
+
+  const graph = VizConfigBuilders.timeseries()
+    .setColor({ mode: 'fixed', fixedColor: getColorByIndex(index) })
+    .build();
+
+  // TODO description
+  // TODO use viz from auto queries
+  // TODO header actions
+
+  return <VizPanel dataProvider={dataProvider} title={metric} viz={graph} />;
```

Comment:
It should be simple to add a wrapper here that sets the CSS `order` property based on the data, this way we can sort the list of tables without having the full list of panels and simply rely on css to reorder the elements